beacon_node/network/src/sync/block_lookups/mod.rs

```diff
@@ -1,91 +1,52 @@
-use self::parent_lookup::PARENT_FAIL_TOLERANCE;
-use self::parent_lookup::{ParentLookup, ParentVerifyError};
-use self::single_block_lookup::{LookupVerifyError, SingleBlockLookup};
+use self::parent_lookup::ParentVerifyError;
+use self::single_block_lookup::SingleBlockLookup;
 use super::manager::BlockProcessingResult;
 use super::BatchProcessResult;
-use super::{
-    manager::{BlockProcessType, Id},
-    network_context::SyncNetworkContext,
-};
+use super::{manager::BlockProcessType, network_context::SyncNetworkContext};
 use crate::metrics;
 use crate::network_beacon_processor::ChainSegmentProcessId;
-use crate::sync::block_lookups::single_block_lookup::LookupId;
+use crate::sync::block_lookups::common::LookupType;
+use crate::sync::block_lookups::parent_lookup::{ParentLookup, RequestError};
+use crate::sync::block_lookups::single_block_lookup::{
+    CachedChild, LookupRequestError, LookupVerifyError,
+};
+use crate::sync::manager::{Id, SingleLookupReqId};
 use beacon_chain::block_verification_types::{AsBlock, RpcBlock};
-use beacon_chain::data_availability_checker::{AvailabilityCheckError, DataAvailabilityChecker};
+use beacon_chain::data_availability_checker::DataAvailabilityChecker;
+use beacon_chain::validator_monitor::timestamp_now;
 use beacon_chain::{AvailabilityProcessingStatus, BeaconChainTypes, BlockError};
+pub use common::Current;
+pub use common::Lookup;
+pub use common::Parent;
+pub use common::RequestState;
+use fnv::FnvHashMap;
 use lighthouse_network::rpc::RPCError;
 use lighthouse_network::{PeerAction, PeerId};
 use lru_cache::LRUTimeCache;
-pub use single_block_lookup::UnknownParentComponents;
+pub use single_block_lookup::CachedChildComponents;
+pub use single_block_lookup::{BlobRequestState, BlockRequestState};
 use slog::{debug, error, trace, warn, Logger};
 use smallvec::SmallVec;
 use std::collections::HashMap;
 use std::fmt::Debug;
 use std::sync::Arc;
 use std::time::Duration;
-use store::{Hash256, SignedBeaconBlock};
+use store::Hash256;
 use strum::Display;
 use types::blob_sidecar::FixedBlobSidecarList;
-use types::{BlobSidecar, Slot};
+use types::Slot;
 
+pub mod common;
 pub(crate) mod delayed_lookup;
 mod parent_lookup;
 mod single_block_lookup;
 #[cfg(test)]
 mod tests;
 
 pub type DownloadedBlocks<T> = (Hash256, RpcBlock<T>);
```

Comment:
Not part of this PR, but just realized this is a single block, perhaps rename this to `DownloadedBlock`? (can also leave it  for later though)